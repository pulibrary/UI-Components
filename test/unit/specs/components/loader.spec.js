import { createLocalVue, mount } from "@vue/test-utils"
import Loader from "@/elements/Loader.vue"

// create an extended `Vue` constructor
const localVue = createLocalVue()

describe("Loader.vue", () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Loader, {
      localVue,
    })
  })

  it("should be full screen when fullscreen prop is true", () => {
    const overlay = wrapper.find(".overlay")
    expect(overlay.classes()).not.toContain("fullscreen")
    wrapper.setProps({ fullscreen: true })
    const fs_overlay = wrapper.find(".overlay")
    expect(fs_overlay.classes()).toContain("fullscreen")
  })

  it("should default to medium size", () => {
    const loader = wrapper.find(".loader")
    expect(loader.attributes().size).toBe("medium")
  })

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
