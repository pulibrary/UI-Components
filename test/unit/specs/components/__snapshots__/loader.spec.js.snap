// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.vue has the expected html structure 1`] = `
<div
  class="overlay"
>
  <div
    class="loader"
    size="medium"
  />
</div>
`;
